File: 프로그래머스/1/42840. 모의고사/모의고사.py
import math

def solution(answers):
    lst_1 = [1, 2, 3, 4, 5] * math.ceil(len(answers) / 5)
    lst_2 = [2, 1, 2, 3, 2, 4, 2, 5] * math.ceil(len(answers) / 8)
    lst_3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] * math.ceil(len(answers) / 10)
    
    counts = {1: 0, 2: 0, 3: 0}
    
    for i in range(len((answers))):
        if answers[i] == lst_1[i]:
            counts[1] += 1
        if answers[i] == lst_2[i]:
            counts[2] += 1
        if answers[i] == lst_3[i]:
            counts[3] += 1
    
    max_value = max(counts.values())
    result = [key for key, value in counts.items() if value == max_value]
    result.sort()

    return result            
        

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const lst_1 = [1, 2, 3, 4, 5];
    const lst_2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const lst_3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    const counts = [0, 0, 0];
    
    for (let i = 0; i < answers.length; i++) {
        if (answers[i] === lst_1[i % lst_1.length]) counts[0]++;
        if (answers[i] === lst_2[i % lst_2.length]) counts[1]++;
        if (answers[i] === lst_3[i % lst_3.length]) counts[2]++;
    }
    
    const max_value = Math.max(...counts);
    
    return counts
        .map((value, idx) => value === max_value ? idx + 1 : null)
        .filter(value => value != null);
}
